pytest_httpserver/httpserver.py

```diff
@@ -675,6 +724,61 @@ def dispatch(self, request: Request) -> Response:
 
         return response
 
+    def _waiting_result_set(self, value: bool) -> None:
+        """Set waiting_result
+
+        Setting is implemented as putting value to queue without waiting. If queue is full we simply ignore the
+        exception, because that means that waiting_result was already set, but not read.
+        """
+        with suppress(queue.Full):
+            self._waiting_result.put_nowait(value)
+
+    def _waiting_result_update(self) -> None:
+        if not self.oneshot_handlers and not self.ordered_handlers:
+            self._waiting_result_set(True)
+
+    @contextmanager
+    def wait(self, raise_assertions: Optional[bool] = None, stop_on_nohandler: Optional[bool] = None,
+             timeout: Optional[float] = None):
+        """Context manager to wait until the first of following event occurs: all ordered and oneshot handlers were
+        executed, unexpected request was received (if `stop_on_nohandler` is set to `True`), or time was out
+
+        :param raise_assertions: whether raise assertions on unexpected request or timeout or not
+        :param stop_on_nohandler: whether stop on unexpected request or not
+        :param timeout: time (in seconds) until time is out
+
+        Example:
+            def test_wait(httpserver):
+                httpserver.expect_oneshot_request('/').respond_with_data('OK')
+                with httpserver.wait(raise_assertions=False, stop_on_nohandler=False, timeout=1) as waiting:
+                    requests.get(httpserver.url_for('/'))
+                # `waiting` is :py:class:`Waiting`
+                assert waiting.result
+                print('Processing time: {} - {} ({})'.format(waiting.start, waiting.stop, waiting.time))
+        """
+        self._waiting_settings.raise_assertions = (
```

Comment:
It is maybe just me but I found these ternaries hard to read and I tend to use normal ifs if it would take more space.
I wouldn't say that it is super critical but I wanted to let you know.

Reply:
See another comment on this.